coder_sniffer/Drupal/Sniffs/Commenting/DeprecatedSniff.php

```diff
@@ -75,20 +74,72 @@ public function process(File $phpcsFile, $stackPtr)
-        // @deprecated in %in-version% and will be removed from %removal-version%. %extra-info%.
-        // Use (?U) 'ungreedy' before the version so that only the text up to
-        // the first '. ' is matched, as there may be more than one sentence in
-        // the extra-info part.
-        $matches = [];
-        preg_match('/in (.+) and is removed from (?U)(.+)\. (.+)$/', $depText, $matches);
+        // @deprecated in %in-version% and is removed from %removal-version%. %extra-info%.
+        $standardFormat = "@deprecated in %%deprecation-version%% and is removed from %%removal-version%%. %%extra-info%%.";
+
+        // Use (?U) 'ungreedy' before the removal-version so that only the text
+        // up to the first dot+space is matched, as there may be more than one
+        // sentence in the extra-info part.
+        $fullText = trim(implode(' ', $textItems));
+        $matches  = [];
+        preg_match('/^in (.+) and is removed from (?U)(.+)(?:\. | |\.$|$)(.*)$/', $fullText, $matches);
         // There should be 4 items in $matches: 0 is full text, 1 = in-version,
-        // 2 = removal-version, 3 = extra-info.
+        // 2 = removal-version, 3 = extra-info (can be blank at this stage).
         if (count($matches) !== 4) {
-            $error = "The text '@deprecated %s' does not match the standard format: @deprecated in %%deprecation-version%% and is removed from %%removal-version%%. %%extra-info%%.";
-            $phpcsFile->addError($error, $stackPtr, 'IncorrectTextLayout', [$depText]);
+            // The full text does not match the standard. Try to find fixes by
+            // testing with a relaxed set of criteria, based on common
+            // formatting variations. This is designed for Core fixes.
+            $error = "The text '@deprecated %s' does not match the standard format: ".$standardFormat;
+            // All of the standard text should be on the first comment line, so
+            // try to match with common formatting errors to allow an automatic
+            // fix. If not possible then report a normal error.
+            $matchesFix = [];
+            $fix        = null;
+            if (count($textItems) > 0) {
+                // Get just the first line of the text.
+                $key   = array_keys($textItems)[0];
+                $text1 = $textItems[$key];
+                preg_match('/^(.*)(as of|in) (drupal|)( |:|)+([\d\.\-xdev\?]+)(,| |. |)(.*)(removed|removal)([ |from|before|in|the]*) (drupal|)( |:|)([\d\-\.xdev]+)( |,|$)+(?:release|)(?:[\.,])*(.*)$/i', $text1, $matchesFix);
+
+                if (count($matchesFix) >= 12) {
+                    // This problem is a drupal core file and is fixable.
+                    if (empty($matchesFix[1]) === false) {
+                        // Verify that its acceptable to remove the text in [1].
+                        echo('>>> File: '.$phpcsFile->path.' line '.$tokens[($stackPtr)]['line'].PHP_EOL);
```

Comment:
hm, I think we should not use echo() in Coder. All output should be done with addError() or addWarning().

Or this for debugging? Then we should hide this behind $this->debug check. See ScopeIndentSniff for example.

Reply:
It was for debug, and useful to keep in, so I will use your hint about $this->debug. Is it ok to call the runtime parameter `deprecated_debug` following the similarly named `scope_indent_debug` ?